kratos/utilities/read_materials_utility.cpp

```diff
@@ -211,13 +203,113 @@ void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
             table.insert(table_param["data"][i][0].GetDouble(),
                          table_param["data"][i][1].GetDouble());
         }
-        p_prop->SetTable(input_var, output_var, table);
+        pNewProperty->SetTable(input_var, output_var, table);
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::CreateSubProperties(
+    ModelPart& rModelPart,
+    Parameters Data,
+    Properties::Pointer& pNewProperty
+    )
+{
+    if (Data.Has("sub_properties")) {
+
+        PointerVectorSet<Properties, IndexedObject> list_sub_properties;
+
+        const std::size_t number_of_subproperties = Data["sub_properties"].size();
+
+        // We do a check of ids
+        for(std::size_t i_sub_prop=0; i_sub_prop < number_of_subproperties; ++i_sub_prop) {
+            const int sub_property_id = Data["sub_properties"][i_sub_prop]["properties_id"].GetInt();
+            KRATOS_ERROR_IF(sub_property_id == static_cast<int>(pNewProperty->Id())) << "You cannot assign to a property a subproperty with the same Id" << std::endl;
+        }
+
+        // We assign the subproperties now
+        for(std::size_t i_sub_prop=0; i_sub_prop < number_of_subproperties; ++i_sub_prop) {
+            // Copy of the current parameters
+            Parameters sub_prop = Data["sub_properties"][i_sub_prop];
+
+            // We get the subproperty id
+            const int sub_property_id = sub_prop["properties_id"].GetInt();
+
+            // We check if already defined
+            const bool already_defined = rModelPart.HasProperties(sub_property_id);
```

Comment:
It should be checked in root modelpart (or at least in the modelpart above)

Reply:
I cannot reply you the previous messages, please create all the comments as this

Reply:
Can you confirm me please that the property should be in the root model part?

Reply:
a property that belongs to a given submodelpart must also belong to its father, and then surely to the top of the tree.

Having said this, accessing directly to the root does not guarantee that this rule is met, so we should be careful about this...

Reply:
It is not verified, @josep-m-carbonell already pointed that some time ago. We should check that first

Reply:
so, as a minimum we need to ensure that this function provides that guarantee

Reply:
Agree, but that can be added in a independent PR, to not increase complexity in this one

Reply:
actually i just verified, and the pGetProperties adds it to all the hierarchy, so this should be resolved.

Only thing is that the warning just above may not be needed